### PR TITLE
Adds 'disabled' class to containing label 

### PR DESCRIPTION
**Pull request by jakemauer**
I figured I'd open a PR as the place to discuss if it's worth adding just this feature or providing a more robust way of controlling label classes.

### CODE REVIEW

File: addon/components/radio-button.js [hunk 1]
```diff
@@ -27,6 +27,7 @@ export default RadioButtonBase.extend({
       );
       this.set('attributeBindings', updatedAttrs);
       this.get('classNameBindings').pushObject('checked');
+      this.get('classNameBindings').pushObject('disabled');
```
Comment (jakemauer) [inline addon/components/radio-button.js]: Turns out you *do* need this line for it to work in an app, and not just in the unit tests. I'm not sure how to test for this now.

**Comment by raycohen** [comment]
ah, the `addon/components/labeled-radio-button.js` file is actually a separate component from `radio-button`.  We aren't documenting it as I wasn't really expect people to use it.  So the first commit that changes that will have no effect if you use `{{#radio-button}}...{{/radio-button}}` in your template.  You *can* use `{{#labeled-radio-button}}...{{/labeled-radio-button}}` and that gives you the labeled form of radio-button without executing `wrapInLabelIfUsedAsBlock`.

**Comment by raycohen** [comment]
here's an idea for an api

```
{{#radio-button disabled-class=true}}  // puts `disabled` class on the label element when disabled

{{#radio-button disabled-class="your-disabled-class"}} // puts "your-disabled-class" on the label when disabled
```
The biggest question I have is if `disabled-class=true` should be the default behavior.  That would match what we did with the `checked` class, but it is also a potentially breaking change for apps currently using ember-radio-button.

**Comment by raycohen** [comment]
@jakemauer After discussing with a few people, I'm leaning towards moving forward with `disabled-class="your-string"` approach, with the default behavior being no class when in the disabled state.  Any thoughts on that?

**Comment by raycohen** [comment]
this has fallen behind master.  we no longer use `classNameBindings`.

I think that adding a `disabled` classname to the label could be accomplished by subclassing the radio-button component and overriding the `joinedClassNames` property to do additionally check `this.get('disabled')` and add to the result if necessary.

We could also add `ember-radio-button-disabled` to the label when it is disabled and that wouldn't carry the risk of impacting styles for existing users.  Are you interested in rebooting this PR with that as a goal?

**Comment by raycohen** [comment]
closing as inactive

**Pull request closed without merging**